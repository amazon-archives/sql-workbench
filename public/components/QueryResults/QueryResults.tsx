/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from "react";
import {
  SortableProperties,
  SortableProperty
} from "@elastic/eui/lib/services";
import {
  EuiPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiTab,
  EuiTabs,
  EuiPopover,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiHorizontalRule,
  EuiSearchBar,
  Pager,
  EuiIcon
} from "@elastic/eui/lib";
import "brace/mode/mysql";
import "brace/mode/json";
import "../../ace-themes/sql_console";
import {
  QueryResult,
  QueryMessage,
  Tab,
  ResponseDetail,
  ItemIdToExpandedRowMap
} from "../Main/main";
import QueryResultsBody from "./QueryResultsBody";
import { getQueryIndex, needsScrolling } from "../../utils/utils";
import {
  DEFAULT_NUM_RECORDS_PER_PAGE,
  MESSAGE_TAB_LABEL,
  TAB_CONTAINER_ID
} from "../../utils/constants";

interface QueryResultsProps {
  queries: string[];
  queryResults: ResponseDetail<QueryResult>[];
  messages: QueryMessage[];
  selectedTabName: string;
  selectedTabId: string;
  searchQuery: string;
  onSelectedTabIdChange: (tab: Tab) => void;
  onQueryChange: (query: string) => void;
  updateExpandedMap: (map: ItemIdToExpandedRowMap) => void;
  itemIdToExpandedRowMap: ItemIdToExpandedRowMap;
  queryResultsRaw: ResponseDetail<string>[];
  queryResultsJDBC: Array<ResponseDetail<string>>;
  queryResultsCSV: Array<ResponseDetail<string>>;
}

interface QueryResultsState {
  isPopoverOpen: boolean;
  tabsOverflow: boolean;
  itemsPerPage: number;
}

class QueryResults extends React.Component<
  QueryResultsProps,
  QueryResultsState
> {
  public sortableColumns: Array<SortableProperty<string>>;
  public sortableProperties: SortableProperties;
  public sortedColumn: string;
  public tabNames: string[];
  public pager: Pager;

  constructor(props: QueryResultsProps) {
    super(props);

    this.state = {
      isPopoverOpen: false,
      tabsOverflow: false,
      itemsPerPage: DEFAULT_NUM_RECORDS_PER_PAGE
    };

    this.sortableColumns = [];
    this.sortedColumn = "";
    this.sortableProperties = new SortableProperties(
      [
        {
          name: "",
          getValue: item => "",
          isAscending: true
        }
      ],
      ""
    );

    this.tabNames = [];
    this.pager = new Pager(0, this.state.itemsPerPage);
  }

  componentDidUpdate() {
    const showArrow = needsScrolling("tabsContainer");
    if (showArrow !== this.state.tabsOverflow) {
      this.setState({ tabsOverflow: showArrow });
    }
  }
  // Actions for Tabs Button
  showTabsMenu = (): void => {
    this.setState(prevState => ({
      isPopoverOpen: !prevState.isPopoverOpen
    }));
  };

  slideTabsRight = (): void => {
    document.getElementById(TAB_CONTAINER_ID).scrollBy(50, 0);
  };

  slideTabsLeft = (): void => {
    document.getElementById(TAB_CONTAINER_ID).scrollBy(-50, 0);
  };

  closePopover = (): void => {
    this.setState({
      isPopoverOpen: false
    });
  };

  onChangeItemsPerPage = itemsPerPage => {
    this.pager.setItemsPerPage(itemsPerPage);
    this.setState({
      itemsPerPage
    });
  };

  onChangePage = pageIndex => {
    this.pager.goToPageIndex(pageIndex);
    this.setState({});
  };

  updatePagination(totalItemsCount: number): void {
    this.pager.setTotalItems(totalItemsCount);
  }

  // Update SORTABLE COLUMNS - All columns
  updateSortableColumns(queryResultsSelected: QueryResult): void {
    if (this.sortableColumns.length === 0) {
      queryResultsSelected.fields.map((field: string) => {
        this.sortableColumns.push({
          name: field,
          getValue: item => item[field],
          isAscending: true
        });
      });
      this.sortedColumn =
        this.sortableColumns.length > 0 ? this.sortableColumns[0].name : "";
      this.sortableProperties = new SortableProperties(
        this.sortableColumns,
        this.sortedColumn
      );
    }
  }

  onSort = prop => {
    this.sortableProperties.sortOn(prop);
    this.sortedColumn = prop;
    this.setState({});
  };

  getQueryResult = (
    queryResults: ResponseDetail<QueryResult>[],
    selectedTabId: string
  ): QueryResult | null => {
    const selectedIndex: number = parseInt(selectedTabId);
    if (!Number.isNaN(selectedIndex)) {
      const queryResultResponseDetail: ResponseDetail<QueryResult> =
        queryResults[selectedIndex];
      return queryResultResponseDetail && queryResultResponseDetail.fulfilled
        ? queryResultResponseDetail.data
        : null;
    }
    return null;
  };

  renderTabs(): Tab[] {
    const tabs = [
      {
        id: MESSAGE_TAB_LABEL,
        name: MESSAGE_TAB_LABEL,
        disabled: false
      }
    ];

    this.tabNames = [];
    if (this.props.queryResults) {
      for (let i = 0; i < this.props.queryResults.length; i += 1) {
        const tabName = getQueryIndex(this.props.queries[i]);
        this.tabNames.push(tabName);
        if (this.props.queryResults[i].fulfilled) {
          tabs.push({
            id: i.toString(),
            name: tabName,
            disabled: false
          });
        }
      }
    }
    return tabs;
  }

  render() {
    // Update PAGINATION  and SORTABLE columns
    const queryResultSelected = this.props.queryResults
      ? this.getQueryResult(this.props.queryResults, this.props.selectedTabId)
      : undefined;

    if (queryResultSelected) {
      const matchingItems: object[] = this.props.searchQuery
        ? EuiSearchBar.Query.execute(
            this.props.searchQuery,
            queryResultSelected.records
          )
        : queryResultSelected.records;
      this.updatePagination(matchingItems.length);
      this.updateSortableColumns(queryResultSelected);
    }

    // Action button with list of tabs, TODO: disable tabArrowRight and tabArrowLeft when no more scrolling is possible
    const tabArrowDown = (
      <EuiIcon onClick={this.showTabsMenu} type={"arrowDown"} />
    );
    const tabArrowRight = (
      <EuiIcon
        onClick={this.slideTabsRight}
        type={"arrowRight"}
        disabled={false}
      />
    );
    const tabArrowLeft = (
      <EuiIcon
        onClick={this.slideTabsLeft}
        type={"arrowLeft"}
        disabled={false}
      />
    );
    const tabs: Tab[] = this.renderTabs();
    const tabsItems = tabs.map((tab, index) => (
      <EuiContextMenuItem
        key="10 rows"
        icon="empty"
        onClick={() => {
          this.closePopover();
          this.pager.goToPageIndex(0);
          this.sortableColumns = [];
          this.props.onSelectedTabIdChange(tab);
        }}
      >
        {tab.name}
      </EuiContextMenuItem>
    ));

    const tabsButtons = tabs.map((tab, index) => (
      <EuiTab
        onClick={() => {
          this.pager.goToPageIndex(0);
          this.sortableColumns = [];
          this.props.onSelectedTabIdChange(tab);
        }}
        isSelected={tab.id === this.props.selectedTabId}
        disabled={tab.disabled}
        key={index}
      >
        {tab.name}
      </EuiTab>
    ));

    return (
      <EuiPanel className="query-result-container" paddingSize="none">
        <EuiFlexGroup
          style={{
            padding: "10px",
            "border-top-color": "#69707d",
            "border-left-color": "#69707d",
            "border-right-color": "#69707d"
          }}
        >
          {/*ARROW LEFT*/}
          {this.state.tabsOverflow && (
            <div className="tab-arrow-down-container">
              <EuiFlexItem grow={false}>{tabArrowLeft}</EuiFlexItem>
            </div>
          )}

          {/*TABS*/}
          <EuiFlexGroup
            className="tabs-container"
            alignItems="center"
            gutterSize="s"
            id="tabsContainer"
          >
            <EuiFlexItem
              style={{ marginTop: "8px", marginBottom: "18px" }}
              grow={false}
            >
              <EuiTabs>{tabsButtons}</EuiTabs>
            </EuiFlexItem>
          </EuiFlexGroup>

          {/*ARROW RIGHT and DOWN*/}
          {this.state.tabsOverflow && (
            <div className="tab-arrow-down-container">
              <EuiFlexGroup>
                <EuiFlexItem grow={false}>{tabArrowRight}</EuiFlexItem>
                <EuiFlexItem grow={false}>
                  <EuiPopover
                    id="singlePanel"
                    button={tabArrowDown}
                    isOpen={this.state.isPopoverOpen}
                    closePopover={this.closePopover}
                    panelPaddingSize="none"
                    anchorPosition="downLeft"
                  >
                    <EuiContextMenuPanel items={tabsItems} />
                  </EuiPopover>
                </EuiFlexItem>
              </EuiFlexGroup>
            </div>
          )}
        </EuiFlexGroup>

        <EuiHorizontalRule margin="none" />

        {/*RESULTS TABLE*/}
        <QueryResultsBody
          selectedTabId={this.props.selectedTabId}
          selectedTabName={this.props.selectedTabName}
          tabNames={this.tabNames}
          queryResultSelected={queryResultSelected}
          queryResultsRaw={this.props.queryResultsRaw}
          queryResultsJDBC={this.props.queryResultsJDBC}
          queryResultsCSV={this.props.queryResultsCSV}
          messages={this.props.messages}
          searchQuery={this.props.searchQuery}
          onQueryChange={this.props.onQueryChange}
          pager={this.pager}
          itemsPerPage={this.state.itemsPerPage}
          firstItemIndex={this.pager.getFirstItemIndex()}
          lastItemIndex={this.pager.getLastItemIndex()}
          onChangeItemsPerPage={this.onChangeItemsPerPage}
          onChangePage={this.onChangePage}
          onSort={this.onSort}
          sortedColumn={this.sortedColumn}
          sortableProperties={this.sortableProperties}
          itemIdToExpandedRowMap={this.props.itemIdToExpandedRowMap}
          updateExpandedMap={this.props.updateExpandedMap}
        />
      </EuiPanel>
    );
  }
}
export default QueryResults;
