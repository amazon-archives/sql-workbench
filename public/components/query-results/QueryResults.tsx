/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */
 
import React, { Component } from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import {
  EuiButton,
  EuiPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiTab,
  EuiTabs,
  EuiPopover,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiHorizontalRule,
  EuiSearchBar,
  Pager,
  EuiContextMenu,
} from '@elastic/eui/lib';
import 'brace/mode/mysql';
import 'brace/mode/json';
import '../../ace-themes/sql_console';
import { QueryResult, QueryMessage, Tab, ResponseDetail} from '../main/main';
import QueryResultsBody from './QueryResultsBody';
import { getQueryIndex } from '../utils/utils';
import { SortableProperties, SortableProperty } from '@elastic/eui/lib/services';

interface QueryResultsProps {
  queries: string[];
  queryResults: ResponseDetail<QueryResult>[];
  message: QueryMessage[];
  selectedTabName: PropTypes.string.isRequired;
  selectedTabId: PropTypes.string.isRequired; 
  searchQuery: PropTypes.string.isRequired;
  onSelectedTabIdChange: (tab: Tab) => void;
  onQueryChange: (query: object) => void;
  updateExpandedMap: (map:object) => void;
  itemIdToExpandedRowMap :  object;
  queryResultsRaw: ResponseDetail<string>[];
  queryResultsJDBC: ResponseDetail<string>[];
  queryResultsCSV: ResponseDetail<string>[];
}

interface QueryResultsState {
  isPopoverOpen: boolean;
  isDownloadPopoverOpen: boolean;
  itemsPerPage: number;
}

function flattenPanelTree(tree, array = []) {
	  array.push(tree);
	  if (tree.items) {
	    tree.items.forEach(item => {
	      if (item.panel) {
	        flattenPanelTree(item.panel, array);
	        item.panel = item.panel.id;
	      }
	    });
	  }	
  return array;
} 

export const MAX_NUM_RECORDS_PER_PAGE = 10;

class QueryResults extends Component<QueryResultsProps, QueryResultsState> {
  public sortableColumns: Array<SortableProperty<string>>;
  public sortableProperties: SortableProperties;
  public sortedColumn: string;
  public tabNames: string[];
  public pager: Pager;
  public selectedTabId: any;
  public panels: string[];
  
  constructor(props: QueryResultsProps) {
    super(props);

    this.state = {
      isPopoverOpen: false,
      isDownloadPopoverOpen: false,
      itemsPerPage: MAX_NUM_RECORDS_PER_PAGE      
    };

   this.sortableColumns = []; 
   this.sortedColumn = '';
   this.sortableProperties = new SortableProperties(
         [{
          name: '',
          getValue: item => '',
          isAscending: true
         }], '');   
   
   this.tabNames = []; 
   this.pager = new Pager(0, this.state.itemsPerPage); 
   
   // Downloads Action button
   const panelTree = {
      id: 0,
      title: 'Download',
      items: [
        {
          name: 'Download JSON',
          onClick: () => { this.onDownloadJSON(); },  
        },
        {
          name: 'Download JDBC',
          onClick: () => { this.onDownloadJDBC(); },
        },
        {
          name: 'Download CVS',
          onClick: () => { this.onDownloadCSV(); },
        },
      ],
    };  
    
    this.panels = flattenPanelTree(panelTree);
  }
  
  // Actions for Downloads Button
  onDownloadButtonClick = (): void => {
    this.setState(prevState => ({
      isDownloadPopoverOpen: !prevState.isDownloadPopoverOpen,
    }));
  };
  
  closeDownloadPopover = (): void => {
    this.setState({
      isDownloadPopoverOpen: false,
    });
  };
  
  // Actions for Tabs Button
  onButtonClick = (): void => {
    this.setState(prevState => ({
      isPopoverOpen: !prevState.isPopoverOpen,
    }));
  };

  closePopover = (): void => {
    this.setState({
      isPopoverOpen: false,
    });
  };
  
  onChangeItemsPerPage = itemsPerPage => {
    this.pager.setItemsPerPage(itemsPerPage);
    this.setState({
      itemsPerPage,
    });
  }

  onChangePage = pageIndex => {
    this.pager.goToPageIndex(pageIndex);
    this.setState({});
  };
  
  updatePagination (totalItemsCount) {
     this.pager.setTotalItems(totalItemsCount);  
  }
  
  // Update SORTABLE COLUMNS - All columns  
  updateSortableColumns(queryResultsSelected) {
      if(this.sortableColumns.length === 0){
	      queryResultsSelected.fields.map((field: string) =>{ 
	         this.sortableColumns.push({
	          name: field,
	          getValue: item => item[field],
	          isAscending: true
	         });
	        }); 
	      this.sortedColumn =  this.sortableColumns.length > 0 ? this.sortableColumns[0].name : '';
	      this.sortableProperties = new SortableProperties(this.sortableColumns, this.sortedColumn);
	   }   
   } 
   
     
  onSort = prop => {
    this.sortableProperties.sortOn(prop);
    this.sortedColumn = prop;
    this.setState({});
  }
  
  onDownloadJDBC = (): void => {
    if (!Number.isNaN(this.selectedTabId)) {
      const jsonObject = JSON.parse(this.props.queryResultsJDBC[this.props.selectedTabId].data);
      const csvContent = 'data:text/json;charset=utf-8,' + JSON.stringify(jsonObject, undefined, 4);
      const encodedUri = encodeURI(csvContent);
      const link = document.createElement("a");
      link.setAttribute('href', encodedUri);
      link.setAttribute('download', this.props.selectedTabName + "JDBC.json");
      document.body.appendChild(link);
      link.click();
      document.body.removeChild(link);
    }
  }
  
  onDownloadJSON = (): void => {   
    if (!Number.isNaN(this.selectedTabId)) {
      const jsonObject = JSON.parse(this.props.queryResultsRaw[this.props.selectedTabId].data);
      const csvContent = 'data:text/json;charset=utf-8,' + JSON.stringify(jsonObject, undefined, 4);
      const encodedUri = encodeURI(csvContent);
      const link = document.createElement("a");
      link.setAttribute('href', encodedUri);
      link.setAttribute('download', this.props.selectedTabName + ".json");
      document.body.appendChild(link);
      link.click();
      document.body.removeChild(link);
    } 
  }  
  
  onDownloadCSV = (): void => {
    if (!Number.isNaN(this.selectedTabId)) {
      const csvContent: string = 'data:text/csv;charset=utf-8,' + this.props.queryResultsCSV[this.props.selectedTabId].data ;
      const encodedUri = encodeURI(csvContent);
      const link = document.createElement("a");
      link.setAttribute('href', encodedUri);
      link.setAttribute('download', this.props.selectedTabName + ".csv");
      document.body.appendChild(link);
      link.click();
      document.body.removeChild(link);
    }

  }
  
  getQueryResult = (queryResults, selectedTabId): QueryResult => {
    const selectedIndex: number = parseInt(selectedTabId);
    if (!Number.isNaN(selectedIndex)) {
      const queryResultResponseDetail: ResponseDetail<QueryResult> = queryResults[selectedIndex];
      return queryResultResponseDetail && queryResultResponseDetail.fulfilled ? queryResultResponseDetail.data : null;
    }
    return null;
  }
  
  renderTabs() : Tab[]{
    const tabs = [{
        id: 'messages',
        name: 'Messages',
        disabled: false,
    }];
    
    this.tabNames = [];
    if(this.props.queryResults) {
	    for (let i = 0; i < this.props.queryResults.length; i += 1) {
	      const tabName = getQueryIndex(this.props.queries[i]);
	      this.tabNames.push(tabName);
	      if (this.props.queryResults[i].fulfilled) {  
	        tabs.push({
	          id: i.toString(),
	          name: tabName,
	          disabled: false,
	        })
	      }
	    }
	}
    return tabs;
  }

  render() {
      
    const enableDownload: boolean = !Number.isNaN(parseInt(this.props.selectedTabId));
    
    // Update PAGINATION  and SORTABLE columns
    const queryResultSelected = this.props.queryResults ? this.getQueryResult(this.props.queryResults, this.props.selectedTabId) : '';
    
    if( queryResultSelected ) {
      const matchingItems = this.props.searchQuery ? EuiSearchBar.Query.execute(this.props.searchQuery, queryResultSelected.records) : queryResultSelected.records;
      this.updatePagination(matchingItems.length);
      this.updateSortableColumns(queryResultSelected);
    }
    
    // Action button with list of tabs
    const button = (
      <EuiButton    
        onClick={this.onButtonClick}
        iconType="arrowDown"
        iconSide="right"
        disabled={false}
      > Tabs
      </EuiButton>
    );
    
    // Action button with list of downloads
    const downloadsButton = (
      <EuiButton
        iconType="arrowDown"
        iconSide="right"
        onClick={this.onDownloadButtonClick}>
        Download
      </EuiButton>
    );
    
    const tabs: Tab[] = this.renderTabs();
    const showArrows: boolean = tabs.length > 2 ;
  
    const tabsItems = tabs.map((tab, index) => (  
      <EuiContextMenuItem
          key="10 rows"
          icon="empty"
          onClick={() => { this.closePopover(); 
                           this.pager.goToPageIndex(0);
                           this.sortableColumns = []; 
                           this.props.onSelectedTabIdChange(tab) }}
        >
        {tab.name}
      </EuiContextMenuItem>   
    ));
    
    const tabsButtons = tabs.map((tab, index) => (
      <EuiTab
        onClick={() => { this.pager.goToPageIndex(0);
                         this.sortableColumns = []; 
                         this.props.onSelectedTabIdChange(tab) }}
        isSelected={tab.id === this.props.selectedTabId}
        disabled={tab.disabled}
        key={index}
      >
        {tab.name}
      </EuiTab>     
    ));
      
    return <EuiPanel className="query-result-container" paddingSize="none">
      <EuiFlexGroup style={{ padding: '10px', 'border-top-color': '#69707d', 'border-left-color': '#69707d', 'border-right-color': '#69707d'}}>
        <EuiFlexGroup className= "tabs-container" alignItems="center" gutterSize="s">  
          <EuiFlexItem style={{ marginTop: '8px', marginBottom: '18px' }} grow={false}>
            <EuiTabs>
              {tabsButtons}
            </EuiTabs>
          </EuiFlexItem>  
        </EuiFlexGroup>
        
        { showArrows &&
            <div className="download-container"> 
	          <EuiFlexItem grow={false}>
	            <EuiPopover
	              id="singlePanel"
	              button={button}
	              isOpen={this.state.isPopoverOpen}
	              closePopover={this.closePopover}
	              panelPaddingSize="none"
	              anchorPosition="downLeft"
	            >
	              <EuiContextMenuPanel items={tabsItems}
	              />
	            </EuiPopover>
	          </EuiFlexItem></div>
          }
          
         { enableDownload &&
         <div className="download-container">    
           <EuiFlexItem grow={false}>
	            <EuiPopover
	              id="singlePanel"
	              button={downloadsButton}
	              isOpen={this.state.isDownloadPopoverOpen}
	              closePopover={this.closeDownloadPopover}
	              panelPaddingSize="none"
	              anchorPosition="downLeft"
	            >
	              <EuiContextMenu initialPanelId={0} panels={this.panels} />
	            </EuiPopover>
	       </EuiFlexItem> </div>
	     }   
      </EuiFlexGroup>

      <EuiHorizontalRule margin="none" />
      <QueryResultsBody selectedTabId={this.props.selectedTabId} 
                        selectedTabName={this.props.selectedTabName} 
                        tabNames={this.tabNames} 
                        queryResults={this.props.queryResults} 
                        message={this.props.message} 
                        searchQuery={this.props.searchQuery}
                        onQueryChange={this.props.onQueryChange} 
                        pager={this.pager} 
                        itemsPerPage={this.state.itemsPerPage}
                        firstItemIndex = {this.pager.getFirstItemIndex()}
                        lastItemIndex = {this.pager.getLastItemIndex()}
                        onChangeItemsPerPage={this.onChangeItemsPerPage}
                        onChangePage={this.onChangePage}
                        onSort={this.onSort}
                        sortedColumn={this.sortedColumn}
                        sortableProperties={this.sortableProperties}
                        itemIdToExpandedRowMap= {this.props.itemIdToExpandedRowMap} 
	                    updateExpandedMap={this.props.updateExpandedMap}
	                    
      />
    </EuiPanel>;
  }
}

interface QueryResultsContainerProps {
  queries: string[];
  queryResultsRaw: ResponseDetail<string>[];
  queryResultsTable: ResponseDetail<QueryResult>[];
  queryResultsJDBC: ResponseDetail<string>[];
  queryResultsCSV: ResponseDetail<string>[];
  selectedTabName: string;
  selectedTabId: string;
  onSelectedTabIdChange: (tab: Tab) => void;
  searchQuery: string;
  onQueryChange: (query: object) => void;
  updateExpandedMap: (map:object) => void;
  message: QueryMessage[];
  itemIdToExpandedRowMap
}

interface QueryResultsContainerState {
}
  
class QueryResultsContainer extends React.Component<QueryResultsContainerProps, QueryResultsContainerState> {
  constructor(props: QueryResultsContainerProps) {
    super(props);
  }

  render() {
    return <QueryResults
		      queries={this.props.queries}
		      queryResultsRaw={this.props.queryResultsRaw}
		      queryResults={this.props.queryResultsTable}
		      queryResultsJDBC={this.props.queryResultsJDBC}
		      queryResultsCSV={this.props.queryResultsCSV}
		      message={this.props.message}
		      searchQuery={this.props.searchQuery}
		      itemIdToExpandedRowMap={this.props.itemIdToExpandedRowMap}
		      onSelectedTabIdChange={this.props.onSelectedTabIdChange}
		      onQueryChange={this.props.onQueryChange}
		      selectedTabId={this.props.selectedTabId}
		      selectedTabName={this.props.selectedTabName}
		      
		      updateExpandedMap={this.props.updateExpandedMap}
		    />
  }
}


export default QueryResultsContainer;
