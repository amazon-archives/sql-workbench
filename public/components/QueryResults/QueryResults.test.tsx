import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import QueryResults from "./QueryResults";
 
describe("<QueryResults /> spec", () => {
  it("renders the component", () => {
    render(
      <QueryResults
        queries={[]}
        queryResults={[]}
        queryResultsJDBC={[]}
        queryResultsRaw={[]}
        queryResultsCSV={[]}
        message={[]}
        selectedTabId={'testTabId'}
        selectedTabName={'testTabId'}
        onSelectedTabIdChange={(a) => {}}
        itemIdToExpandedRowMap={{}}
        onQueryChange={(a) => {}}
        updateExpandedMap={(a) => {}}
        searchQuery={'testQuery'}
      />
    );
    expect(document.body.children[0]).toMatchSnapshot();
  });
});