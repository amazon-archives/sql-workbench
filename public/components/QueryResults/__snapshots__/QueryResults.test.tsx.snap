// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryResults /> spec renders the component with no data 1`] = `
<div>
  <div
    class="euiPanel query-result-container"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 10px; border-top-color: #69707d; border-left-color: #69707d; border-right-color: #69707d;"
    >
      <div
        class="tab-arrow-down-container"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            class="euiIcon euiIcon--medium"
            data-test-subj="slide-left"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <defs>
              <path
                d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                id="arrow_left-a"
              />
            </defs>
            <use
              fill-rule="nonzero"
              transform="rotate(90 8 8)"
              xlink:href="#arrow_left-a"
            />
          </svg>
        </div>
      </div>
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive tabs-container"
        id="tabsContainer"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
          style="margin-top: 8px; margin-bottom: 18px;"
        >
          <div
            class="euiTabs"
            role="tablist"
          >
            <button
              aria-selected="false"
              class="euiTab"
              role="tab"
              type="button"
            >
              <span
                class="euiTab__content"
              >
                Messages
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="tab-arrow-down-container"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <svg
              class="euiIcon euiIcon--medium"
              data-test-subj="slide-right"
              focusable="false"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs>
                <path
                  d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                  id="arrow_right-a"
                />
              </defs>
              <use
                fill-rule="nonzero"
                transform="matrix(0 1 1 0 0 0)"
                xlink:href="#arrow_right-a"
              />
            </svg>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiPopover euiPopover--anchorDownLeft"
              id="singlePanel"
            >
              <div
                class="euiPopover__anchor"
              >
                <svg
                  class="euiIcon euiIcon--medium"
                  data-test-subj="slide-down"
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                      id="arrow_down-a"
                    />
                  </defs>
                  <use
                    fill-rule="nonzero"
                    xlink:href="#arrow_down-a"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full"
    />
    <div
      class="euiCodeEditorWrapper"
      data-test-subj="codeEditorContainer"
      style="width: 100%;"
    >
      <div
        class="euiCodeEditorKeyboardHint"
        data-test-subj="codeEditorHint"
        id="some_html_id"
        role="button"
        tabindex="0"
      >
        <p
          class="euiText"
        >
          Press Enter to start editing.
        </p>
        <p
          class="euiText"
        >
          When you're done, press Escape to stop editing.
        </p>
      </div>
      <div
        class=" ace_editor ace-sql-console successful-message"
        id="brace-editor"
        style="width: 100%; height: 500px; font-size: 14px;"
      >
        <textarea
          aria-label="Code Editor"
          autocapitalize="off"
          autocorrect="off"
          class="ace_text-input"
          spellcheck="false"
          style="opacity: 0;"
          tabindex="-1"
          wrap="off"
        />
        <div
          aria-hidden="true"
          class="ace_gutter"
        >
          <div
            class="ace_layer ace_gutter-layer ace_folding-enabled"
          />
          <div
            class="ace_gutter-active-line"
            style="display: none;"
          />
        </div>
        <div
          class="ace_scroller"
        >
          <div
            class="ace_content"
          >
            <div
              class="ace_layer ace_print-margin-layer"
            >
              <div
                class="ace_print-margin"
                style="left: 4px; visibility: hidden;"
              />
            </div>
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_text-layer"
              style="padding: 0px 4px;"
            />
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_cursor-layer ace_hidden-cursors"
            >
              <div
                class="ace_cursor"
              />
            </div>
          </div>
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-v"
          style="display: none; width: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="width: 20px;"
          />
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-h"
          style="display: none; height: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="height: 20px;"
          />
        </div>
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
        >
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          />
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          >
            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
